src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1104,7 +1364,7 @@ private bool IsDownstreamOutDefault(Pipe downstreamPipe)
         /// <param name="redirectInput"></param>
         private void CalculateIORedirection(out bool redirectOutput, out bool redirectError, out bool redirectInput)
         {
-            redirectInput = true;
+            redirectInput = this.Command.MyInvocation.PipelineLength > 0;
```

Comment:
The native command may be the first command in the pipeline, and it doesn't need input in that case.